soroban-env-host/src/events/internal.rs

```diff
@@ -156,30 +170,43 @@ impl InternalEventsBuffer {
     /// Converts the internal events into their external representation. This should only be called
     /// either when the host is finished (via `try_finish`), or when an error occurs.
     pub fn externalize(&self, host: &Host) -> Result<Events, HostError> {
-        let vec: Result<Vec<HostEvent>, HostError> = self
-            .vec
-            .iter()
-            .map(|e| match &e.0 {
+        let mut vec: Vec<HostEvent> = vec![];
+        vec.reserve(self.vec.len());
```

Comment:
Shouldn't we used `metered_collect` here?

Reply:
No. This line is intentionally unmetered, because metering is taken care of inside the loop (right before `vec.push`) to ensure metering is indifferent of the buffer length.